src/sbin/sympa.pl.in

```diff
@@ -571,6 +575,39 @@ if ($main::options{'dump'} or $main::options{'dump_users'}) {
     }
     exit 0;
 
+} elsif ($main::options{'add_list_admin'}) {
+    unless ($main::options{'email'} and $main::options{'role'}) {
+        print STDERR "Missing email or role option\n";
+        exit 1;
+    }
+
+    my ($listname, $robot_id) = split '@', $main::options{'list'};
+    my $list = Sympa::List->new($listname, $robot_id);
+    unless ($list) {
+        printf STDERR "Incorrect list name %s\n",
+            $main::options{'list'};
+        exit 1;
+    }
```

Comment:
All inputs by users including parameter values should be validated (email, role, profile, ...).

Reply:
Testing parameters are done by in fee59b6.
Note that these parameters are tested in the module, not in sympa.pl. I thought it was better this way.
I still need to change the sympa.pl command to reflect errors.
Ideally, I would use the report.tt2. Still need to gather how to use it.

Reply:
OK. I did not notice the "_report" sub in sympa.pl.. I used it though... So it looks like everything is cool by now.

Reply:
I'm OK.